### PR TITLE
fix: Dashboard crashing

### PR DESCRIPTION
**Pull request by stackchain**
There were some especific cases where the staking dashboard will crash. I.e pool with an empty homepage.

### CODE REVIEW

File: src/TxHistory/TxDetails/TxDetails.tsx [hunk 1]
```diff
@@ -139,7 +139,7 @@ export const TxDetails = () => {
         </View>
       </Screen>
 
-      {addressDetail && (
+      {!!addressDetail && (
```
Comment (wolverineks) [inline src/TxHistory/TxDetails/TxDetails.tsx:142]: how does this change anything?

File: src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ class ModalClassComponent extends React.Component<Props & NavigationHookProp, St
       <RNModal transparent animationType="fade" visible={visible && isFocused} onRequestClose={onRequestClose}>
         <View style={styles.backdrop}>
           <View style={[styles.container, noPadding && styles.noPadding, !!title && styles.withTitle]}>
-            {title && <Text style={styles.title}>{title}</Text>}
+            {title != null && <Text style={styles.title}>{title}</Text>}
```
Comment (wolverineks) [inline src/components/Modal/Modal.tsx]: why display `""`?

File: src/Dashboard/StakePoolInfo.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ export const StakePoolInfo = ({stakePoolId}: {stakePoolId: string}) => {
           </View>
         </View>
 
-        {stakePoolInfo.homepage && (
+        {!!stakePoolInfo.homepage && (
```
Comment (wolverineks) [inline src/Dashboard/StakePoolInfo.tsx:39]: how does this change anything?

File: src/Dashboard/EpochProgress.tsx [hunk 4]
```diff
@@ -35,7 +35,7 @@ export const EpochProgress = ({percentage, currentEpoch, endTime}: Props) => {
           <View style={styles.row}>
             <Text style={styles.label}>{intl.formatMessage(messages.endsInLabel)}:</Text>
             <View style={styles.timeWrapper}>
-              {endTime.d && <Text style={styles.timeBlock}>{endTime.d}</Text>}
+              {endTime.d != null && <Text style={styles.timeBlock}>{endTime.d}</Text>}
```
Comment (wolverineks) [inline src/Dashboard/EpochProgress.tsx:38]: what does this change?

Comment (stackchain) [reply src/Dashboard/EpochProgress.tsx:38]: Text strings must be rendered within a component.

File: src/Dashboard/StakePoolInfo.tsx (hunk 3 above)
Comment (stackchain) [reply src/Dashboard/StakePoolInfo.tsx:39]: Text strings must be rendered within a component.

File: src/TxHistory/TxDetails/TxDetails.tsx (hunk 1 above)
Comment (stackchain) [reply src/TxHistory/TxDetails/TxDetails.tsx:142]: Text strings must be rendered within a component.

File: src/components/Modal/Modal.tsx (hunk 2 above)
Comment (stackchain) [reply src/components/Modal/Modal.tsx]: Yap, let's make it consistent with the styles check.
```suggestion
            {!!title && <Text style={styles.title}>{title}</Text>}
```

**Review by banklesss — APPROVED** [review]

**Comment by wolverineks** [comment]
can you look into eslint rules to prevent this from happening again?
maybe something like https://typescript-eslint.io/rules/strict-boolean-expressions/

**Comment by stackchain** [comment]
> can you look into eslint rules to prevent this from happening again? maybe something like https://typescript-eslint.io/rules/strict-boolean-expressions/

Task created.

**Pull request merged**